### ISSUE TITLE
false positive that should be caught?

### ISSUE THREAD

**Issue by rgetz** [david-a-wheeler/flawfinder#20]
On this line of code:

`sscanf(ptr, "%08" PRIx32, &mask[i - 1]);`

I'm getting a flagged:
```
The scanf() family's %s operation, without a limit specification, permits buffer overflows (CWE-120, CWE-20). 
```
The limit spec is there. `"%08`
The format is defined in: `/usr/include/inttypes.h` as `# define PRIx32		"x"`

The `PRI`  format conversion specifies are a pretty common way of writing portable code now... They are part of the [Open Group](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/inttypes.h.html) as part of Issue 5, released 1997. (So you know these reserved words are not a variable in the code).

I can understand why it should warn about using the `PRI` printing formats when I should be using the `SCN` scanning (should be using `SCNx32`) - but I think the existing fault is wrong. (I think).

**Comment by david-a-wheeler** [comment]
Hmm, this definitely needs some refinement. Since it's in a widely-supported standard, and these symbols are unlikely to have any other meaning, we could certainly tweak flawfinder to deal with these in a better way. Want to write some code? :-).

**Comment by rgetz** [comment]
Yeah, I know enough python to know my python is terrible. (I'm  not very pythonic).

Depends on how many times you want to re-write it. :)
